crates/_support/objkv/tests/snapshot_survives_collection.sh

```
@@ -0,0 +1,101 @@
+echo "1. session A opens REPEATABLE READ and reads"
+# A is one psql fed through a pipe and kept open across the churn. Its output
+# is a file psql flushes at exit, so progress is signalled with \! touch.
+mkfifo "$WORK/a.in"
+psqlx -d postgres -tA <"$WORK/a.in" >"$WORK/a.out" 2>&1 &
+A_PID=$!
+exec 3>"$WORK/a.in"
+a() { printf '%s\n' "$1" >&3; }
+a_reached() {  # a_reached <mark>: true once A has run everything sent before it
+    local i
+    a "\\! touch '$WORK/a.$1'"
+    for i in $(seq 1 240); do [ -e "$WORK/a.$1" ] && return 0; sleep 0.25; done
```

Shellcheck (0.11.0)

[warning] i appears unused. Verify use (or export if used externally).

(SC2034)